Mail routing problem

### Expected behavior
Mail is routed properly

### Observed behavior
I did reprovision my jails to change them to the new lo1 interface and new network range.
However I struggled to get it working again, since the mail is routed wrong.

BTW first I struggled since qmail-delivered didn't have access to mysql due the change in IP

But now this is the result:

```
Oct 12 02:48:20 haraka haraka[2620]: [INFO] [DFA94DF0-FEE0-467E-81F7-7FD78830A32E] [karma] score: 0, good: 56, bad: 1, connections: 62, history: 55
Oct 12 02:48:20 haraka haraka[2620]: [INFO] [DFA94DF0-FEE0-467E-81F7-7FD78830A32E] [p0f] os="Linux 2.2.x-3.x" link_type="generic tunnel or VPN" distance=19 total_conn=1
Oct 12 02:48:22 haraka haraka[2620]: [INFO] [DFA94DF0-FEE0-467E-81F7-7FD78830A32E] [dnsbl] fail:dnsbl.sorbs.net
Oct 12 02:48:22 haraka haraka[2620]: [INFO] [DFA94DF0-FEE0-467E-81F7-7FD78830A32E] [helo.checks] helo_host: mail-qt0-f177.google.com, ips: 209.85.216.177, pass:match_re, bare_ip, dynamic, big_co(not), literal_mismatch, valid_hos
tname, rdns_match, forward_dns, host_mismatch
Oct 12 02:48:22 haraka haraka[2620]: [INFO] [DFA94DF0-FEE0-467E-81F7-7FD78830A32E] [spf] identity=helo ip=209.85.216.177 domain="mail-qt0-f177.google.com" mfrom=<postmaster@mail-qt0-f177.google.com> result=None
Oct 12 02:48:22 haraka haraka[2620]: [INFO] [DFA94DF0-FEE0-467E-81F7-7FD78830A32E] [spf] scope: helo, result: None, domain: mail-qt0-f177.google.com
Oct 12 02:48:22 haraka haraka[2620]: [INFO] [DFA94DF0-FEE0-467E-81F7-7FD78830A32E] [tls] secured: cipher=ECDHE-RSA-AES128-GCM-SHA256 version=TLSv1/SSLv3 verified=false cn="smtp.gmail.com" organization="Google Inc" issuer="Google Inc" expires="Aug  7 00:00:00 2018 GMT" fingerprint=3E:4F:1E:DF:D6:FD:99:13:B5:C1:86:CD:12:30:24:26:B0:10:45:94
Oct 12 02:48:22 haraka haraka[2620]: [INFO] [DFA94DF0-FEE0-467E-81F7-7FD78830A32E] [core]  hook=unrecognized_command plugin=tls function=upgrade_connection params=STARTTLS retval=OK msg=""
Oct 12 02:48:22 haraka haraka[2620]: [INFO] [DFA94DF0-FEE0-467E-81F7-7FD78830A32E] [helo.checks] helo_host: mail-qt0-f177.google.com, ips: 209.85.216.177, multi: true, pass:match_re, bare_ip, dynamic, big_co(not), literal_mismatch, valid_hostname, rdns_match, forward_dns, host_mismatch, host_mismatch
Oct 12 02:48:22 haraka haraka[2620]: [INFO] [DFA94DF0-FEE0-467E-81F7-7FD78830A32E] [spf] identity=helo ip=209.85.216.177 domain="mail-qt0-f177.google.com" mfrom=<postmaster@mail-qt0-f177.google.com> result=None
Oct 12 02:48:22 haraka haraka[2620]: [INFO] [DFA94DF0-FEE0-467E-81F7-7FD78830A32E] [spf] scope: helo, result: None, domain: mail-qt0-f177.google.com
Oct 12 02:48:22 haraka haraka[2620]: [INFO] [DFA94DF0-FEE0-467E-81F7-7FD78830A32E.1] [spf] identity=mfrom ip=209.85.216.177 domain="gmail.com" mfrom=<r.j.lensen@gmail.com> result=Pass
Oct 12 02:48:23 haraka haraka[2620]: [INFO] [DFA94DF0-FEE0-467E-81F7-7FD78830A32E.1] [spf] scope: mfrom, result: Pass, domain: gmail.com
Oct 12 02:48:23 haraka haraka[2620]: [INFO] [DFA94DF0-FEE0-467E-81F7-7FD78830A32E.1] [qmail-deliverable] not local
Oct 12 02:48:23 haraka haraka[2620]: [NOTICE] [DFA94DF0-FEE0-467E-81F7-7FD78830A32E.1] [core] sender <r.j.lensen@gmail.com> code=CONT msg=""
Oct 12 02:48:23 haraka haraka[2620]: [INFO] [DFA94DF0-FEE0-467E-81F7-7FD78830A32E.1] [qmail-deliverable] vpopmail alias
Oct 12 02:48:23 haraka haraka[2620]: [INFO] [DFA94DF0-FEE0-467E-81F7-7FD78830A32E.1] [qmail-deliverable] msg:mail_from.not local, pass:rcpt.vpopmail alias
Oct 12 02:48:23 haraka haraka[2620]: [INFO] [DFA94DF0-FEE0-467E-81F7-7FD78830A32E.1] [core]  hook=rcpt plugin=qmail-deliverable function=hook_rcpt params=<rob@lensen.nu> retval=OK msg=""
Oct 12 02:48:23 haraka haraka[2620]: [NOTICE] [DFA94DF0-FEE0-467E-81F7-7FD78830A32E.1] [core] recipient <rob@lensen.nu> code=OK msg="" sender=r.j.lensen@gmail.com
Oct 12 02:48:23 haraka haraka[2620]: [INFO] [DFA94DF0-FEE0-467E-81F7-7FD78830A32E.1] [avg] { time: '12ms', code: '200', response: '"ok"' }
Oct 12 02:48:31 haraka haraka[2620]: [INFO] [DFA94DF0-FEE0-467E-81F7-7FD78830A32E.1] [rspamd] is_spam: false, is_skipped: false, score: -0.1, required_score: 15, action: no action, TO_DN_NONE: 0, RCVD_COUNT_THREE: 0, FROM_HAS_DN: 0, FROM_EQ_ENVFROM: 0, RCVD_NO_TLS_LAST: 0, MIME_GOOD: -0.1, TO_MATCH_ENVRCPT_ALL: 0, TAGGED_FROM: 0, RCPT_COUNT_ONE: 0, PREVIOUSLY_DELIVERED: 0, time: 8.017
Oct 12 02:48:31 haraka haraka[2620]: [NOTICE] [DFA94DF0-FEE0-467E-81F7-7FD78830A32E.1] [core] message mid="<CAEgN0qJ4MyWh3-8S+0HKvime-83jRA=m_KzPodtXEo7Zih8t+A@mail.gmail.com>" size=2349 rcpts=1/0/0 delay=8.16 code=CONT msg=""
Oct 12 02:48:31 haraka haraka[2620]: [INFO] [-] [core] [outbound] Sending email as a transaction
Oct 12 02:48:31 haraka haraka[2620]: [ERROR] [-] [dkim_sign] skipped: missing dkim private key
Oct 12 02:48:31 haraka haraka[2620]: [INFO] [-] [core] [outbound] Processing domain: lensen.nu
Oct 12 02:48:31 haraka haraka[2620]: [INFO] [DFA94DF0-FEE0-467E-81F7-7FD78830A32E.1] [core]  hook=queue plugin=qmail-deliverable function=hook_queue params="" retval=OK msg="Message Queued"
Oct 12 02:48:31 haraka haraka[2620]: [NOTICE] [DFA94DF0-FEE0-467E-81F7-7FD78830A32E.1] [core] queue code=OK msg="Message Queued (DFA94DF0-FEE0-467E-81F7-7FD78830A32E.1)"
Oct 12 02:48:31 haraka haraka[2620]: [INFO] [DFA94DF0-FEE0-467E-81F7-7FD78830A32E.1] [karma] score: 2, good: 56, bad: 1, connections: 62, history: 55, awards: 131,183,182,232
Oct 12 02:48:31 haraka haraka[2620]: [INFO] [DFA94DF0-FEE0-467E-81F7-7FD78830A32E.1.1] [outbound] Looking up A records for: smtp.bsdfreaks.nl
Oct 12 02:48:31 haraka haraka[2620]: [INFO] [DFA94DF0-FEE0-467E-81F7-7FD78830A32E.1.1] [outbound] Attempting to deliver to: 149.210.213.35:25 (0) (2)
Oct 12 02:48:31 haraka haraka[2620]: [INFO] [-] [core] [outbound] [outbound::25:149.210.213.35:undefined:50] dispense() clients=1 available=0
Oct 12 02:48:31 haraka haraka[2620]: [INFO] [DFA94DF0-FEE0-467E-81F7-7FD78830A32E.1] [karma] score: 2, good: 56, bad: 1, connections: 62, history: 55, awards: 131,183,182,232
Oct 12 02:48:31 haraka haraka[2620]: [NOTICE] [DFA94DF0-FEE0-467E-81F7-7FD78830A32E.1] [core] disconnect ip=209.85.216.177 rdns=mail-qt0-f177.google.com helo=mail-qt0-f177.google.com relay=N early=N esmtp=Y tls=Y pipe=Y errors=0 txns=1 rcpts=1/0/0 msgs=1/0/0 bytes=2349 lr="" time=11.253
Oct 12 02:49:01 haraka haraka[2620]: [ERROR] [-] [core] [outbound] Failed to get pool entry: Outbound connection timed out to 149.210.213.35:25
Oct 12 02:49:01 haraka haraka[2620]: [INFO] [-] [core] [outbound] [outbound::25:149.210.213.35:undefined:50] dispense() clients=0 available=0
Oct 12 02:49:01 haraka haraka[2620]: [INFO] [DFA94DF0-FEE0-467E-81F7-7FD78830A32E.1.1] [outbound] Temp failing 1507769311487_1507769311487_0_2620_OEUr2f_3_haraka for 64 seconds: Tried all MXs
```

Mail coming from gmail for one of my domains (lensen.nu) is accepted but then again, pushed forward to outbound. However it should be routed to the vpopmail jail

I checked qmail-delivered this is responding correct

I finally got it fixed by adding this to mt6-local.conf

```
 local-data: "8.15.16.172.in-addr.arpa PTR smtp.bsdfreaks.nl"
 local-data: "smtp.bsdfreaks.nl A 172.16.15.8"
 local-data: "smtp.bsdfreaks.nl TXT 'v=spf1 a -all"
```
smtp.bsdfreaks.nl is the MX record

I think this shouldn't be the fix...
Meanwhile I did change quite some things... but don't know if it's right

```
 # cat qmail-deliverable.ini
check_outbound=true
host=172.16.15.8
port=8998
```

```
 # cat auth_vpopmaild.ini 
host=172.16.15.8
port=89
```

```
 # less host_list
smtp.bsdfreaks.nl
```

```
 # cat outbound.ini
enable_tls=true
```

```
# cat smtp_forward.ini 
host=172.16.15.8
port=25
```
### Vpopmail

```
 # cat defaultdomain
bsdfreaks.nl
```

```
# cat locals
vpopmail.bsdfreaks.nl
```

```
# cat me
vps1.bsdfreaks.nl
```

```
# cat plusdomain
bsdfreaks.nl
```

```
# cat etc/tcp.smtp
# if the chkuser patch is compiled into qmail,
# CHKUSER_MBXQUOTA rejects messages when the users mailbox quota is filled
127.0.0.1:allow,RELAYCLIENT=""
172.16.15.9:allow,RELAYCLIENT=""
:allow,CHKUSER_MBXQUOTA="99"
```

After change to mt6-local.conf

```
Oct 12 03:50:58 haraka haraka[47647]: [NOTICE] [B45E4EBF-FC45-4F2F-A041-115BA431FB6F] [core] connect ip=40.107.1.94 port=59847 local_ip=:: local_port=25
Oct 12 03:50:58 haraka haraka[47647]: [INFO] [B45E4EBF-FC45-4F2F-A041-115BA431FB6F] [p0f] os="Windows 7 or 8" link_type="Ethernet or modem" distance=15 total_conn=1
Oct 12 03:50:58 haraka haraka[47647]: [INFO] [B45E4EBF-FC45-4F2F-A041-115BA431FB6F] [dnsbl]
Oct 12 03:50:58 haraka haraka[47647]: [INFO] [B45E4EBF-FC45-4F2F-A041-115BA431FB6F] [helo.checks] helo_host: EUR02-HE1-obe.outbound.protection.outlook.com, ips: 2a01:111:f400:7e05::200,213.199.180.175, pass:match_re, bare_ip, dy
namic, big_co(not), literal_mismatch, valid_hostname, rdns_match(org_dom), forward_dns(domain), host_mismatch
Oct 12 03:50:59 haraka haraka[47647]: [INFO] [B45E4EBF-FC45-4F2F-A041-115BA431FB6F] [spf] identity=helo ip=40.107.1.94 domain="EUR02-HE1-obe.outbound.protection.outlook.com" mfrom=<postmaster@EUR02-HE1-obe.outbound.protection.ou
tlook.com> result=Pass
Oct 12 03:50:59 haraka haraka[47647]: [INFO] [B45E4EBF-FC45-4F2F-A041-115BA431FB6F] [spf] scope: helo, result: Pass, domain: EUR02-HE1-obe.outbound.protection.outlook.com
Oct 12 03:50:59 haraka haraka[47647]: [INFO] [B45E4EBF-FC45-4F2F-A041-115BA431FB6F] [tls] secured: cipher=ECDHE-RSA-AES128-SHA256 version=TLSv1/SSLv3 verified=false cn="mail.protection.outlook.com" organization="Microsoft Corpor
ation" issuer="Microsoft Corporation" expires="Apr 30 16:33:37 2018 GMT" fingerprint=6C:D7:E6:0C:21:66:09:E3:AD:5D:FA:E8:7C:A0:DC:54:F3:22:52:AC
Oct 12 03:50:59 haraka haraka[47647]: [INFO] [B45E4EBF-FC45-4F2F-A041-115BA431FB6F] [core]  hook=unrecognized_command plugin=tls function=upgrade_connection params=STARTTLS retval=OK msg=""
Oct 12 03:50:59 haraka haraka[47647]: [INFO] [B45E4EBF-FC45-4F2F-A041-115BA431FB6F] [helo.checks] helo_host: EUR02-HE1-obe.outbound.protection.outlook.com, ips: 2a01:111:f400:7e05::200,213.199.180.175, multi: true, pass:match_re
, bare_ip, dynamic, big_co(not), literal_mismatch, valid_hostname, rdns_match(org_dom), forward_dns(domain), host_mismatch, host_mismatch
Oct 12 03:50:59 haraka haraka[47647]: [INFO] [B45E4EBF-FC45-4F2F-A041-115BA431FB6F] [spf] identity=helo ip=40.107.1.94 domain="EUR02-HE1-obe.outbound.protection.outlook.com" mfrom=<postmaster@EUR02-HE1-obe.outbound.protection.ou
tlook.com> result=Pass
Oct 12 03:50:59 haraka haraka[47647]: [INFO] [B45E4EBF-FC45-4F2F-A041-115BA431FB6F] [spf] scope: helo, result: Pass, domain: EUR02-HE1-obe.outbound.protection.outlook.com
Oct 12 03:51:00 haraka haraka[47647]: [INFO] [B45E4EBF-FC45-4F2F-A041-115BA431FB6F.1] [spf] identity=mfrom ip=40.107.1.94 domain="linde.com" mfrom=<outlook.account@linde.com> result=Pass
Oct 12 03:51:00 haraka haraka[47647]: [INFO] [B45E4EBF-FC45-4F2F-A041-115BA431FB6F.1] [spf] scope: mfrom, result: Pass, domain: linde.com
Oct 12 03:51:00 haraka haraka[47647]: [INFO] [B45E4EBF-FC45-4F2F-A041-115BA431FB6F.1] [qmail-deliverable] not local
Oct 12 03:51:00 haraka haraka[47647]: [NOTICE] [B45E4EBF-FC45-4F2F-A041-115BA431FB6F.1] [core] sender <outlook.account@linde.com> code=CONT msg=""
Oct 12 03:51:00 haraka haraka[47647]: [INFO] [B45E4EBF-FC45-4F2F-A041-115BA431FB6F.1] [qmail-deliverable] vpopmail alias
Oct 12 03:51:00 haraka haraka[47647]: [INFO] [B45E4EBF-FC45-4F2F-A041-115BA431FB6F.1] [qmail-deliverable] msg:mail_from.not local, pass:rcpt.vpopmail alias
Oct 12 03:51:00 haraka haraka[47647]: [INFO] [B45E4EBF-FC45-4F2F-A041-115BA431FB6F.1] [core]  hook=rcpt plugin=qmail-deliverable function=hook_rcpt params=<internal_account@lensen.nu> retval=OK msg=""
Oct 12 03:51:00 haraka haraka[47647]: [NOTICE] [B45E4EBF-FC45-4F2F-A041-115BA431FB6F.1] [core] recipient <internal_account@lensen.nu> code=OK msg="" sender=outlook.account@linde.com
Oct 12 03:51:00 haraka haraka[47647]: [INFO] [B45E4EBF-FC45-4F2F-A041-115BA431FB6F.1] [avg] { time: '9ms', code: '200', response: '"ok"' }
Oct 12 03:51:08 haraka haraka[47647]: [INFO] [B45E4EBF-FC45-4F2F-A041-115BA431FB6F.1] [rspamd] is_spam: false, is_skipped: false, score: 5, required_score: 15, action: greylist, RCVD_COUNT_THREE: 0, TO_DN_EQ_ADDR_ALL: 0, FROM_EQ
_ENVFROM: 0, FROM_HAS_DN: 0, RCVD_NO_TLS_LAST: 0, MIME_GOOD: -0.1, COMPROMISED_ACCT_BULK: 3, TO_MATCH_ENVRCPT_ALL: 0, RCPT_COUNT_ONE: 0, MID_RHS_MATCH_FROM: 0, HAS_XOIP: 0, MIME_BASE64_TEXT: 0.1, DCC_BULK: 2, time: 8.047
Oct 12 03:51:08 haraka haraka[47647]: [NOTICE] [B45E4EBF-FC45-4F2F-A041-115BA431FB6F.1] [core] message mid=<DA9431EA-F95B-4639-A5CD-EBC2ED38BF42@linde.com> size=7140 rcpts=1/0/0 delay=8.096 code=CONT msg=""
Oct 12 03:51:08 haraka haraka[47647]: [INFO] [-] [core] [outbound] Sending email as a transaction
Oct 12 03:51:08 haraka haraka[47647]: [INFO] [-] [core] [outbound] Processing domain: lensen.nu
Oct 12 03:51:08 haraka haraka[47647]: [INFO] [B45E4EBF-FC45-4F2F-A041-115BA431FB6F.1] [core]  hook=queue plugin=qmail-deliverable function=hook_queue params="" retval=OK msg="Message Queued"
Oct 12 03:51:08 haraka haraka[47647]: [NOTICE] [B45E4EBF-FC45-4F2F-A041-115BA431FB6F.1] [core] queue code=OK msg="Message Queued (B45E4EBF-FC45-4F2F-A041-115BA431FB6F.1)"
Oct 12 03:51:08 haraka haraka[47647]: [INFO] [B45E4EBF-FC45-4F2F-A041-115BA431FB6F.1] [karma] score: 0, awards: 131,183,182,231
Oct 12 03:51:08 haraka haraka[47647]: [INFO] [B45E4EBF-FC45-4F2F-A041-115BA431FB6F.1.1] [outbound] Looking up A records for: smtp.bsdfreaks.nl
Oct 12 03:51:08 haraka haraka[47647]: [INFO] [B45E4EBF-FC45-4F2F-A041-115BA431FB6F.1.1] [outbound] Attempting to deliver to: 172.16.15.8:25 (0) (0)
Oct 12 03:51:08 haraka haraka[47647]: [INFO] [-] [core] [outbound] [outbound::25:172.16.15.8:undefined:50] dispense() clients=1 available=0
Oct 12 03:51:08 haraka haraka[47647]: [INFO] [B45E4EBF-FC45-4F2F-A041-115BA431FB6F.1] [karma] score: 0, awards: 131,183,182,231
Oct 12 03:51:08 haraka haraka[47647]: [NOTICE] [B45E4EBF-FC45-4F2F-A041-115BA431FB6F.1] [core] disconnect ip=40.107.1.94 rdns=mail-eopbgr10094.outbound.protection.outlook.com helo=EUR02-HE1-obe.outbound.protection.outlook.com relay=N early=N esmtp=Y tls=Y pipe=Y errors=0 txns=1 rcpts=1/0/0 msgs=1/0/0 bytes=7140 lr="" time=10.401
Oct 12 03:51:08 haraka haraka[47647]: [NOTICE] [B45E4EBF-FC45-4F2F-A041-115BA431FB6F.1.1] [outbound]  delivered file=1507773068356_1507773068356_0_47647_Q2YB1i_4_haraka domain=lensen.nu host=smtp.bsdfreaks.nl ip=172.16.15.8 port=25 mode=SMTP tls=N auth=N response="ok 1507773068 qp 50774" delay=0.142 fails=0 rcpts=1/0/0
```

--- a/include/vpopmail.sh
+++ b/include/vpopmail.sh
@@ -57,7 +57,7 @@ install_qmail()
 		local _qmdir="$STAGE_MNT/var/qmail/$_cdir"
 		if [ -d "$_qmdir" ]; then
 			tell_status "rm -rf $_qmdir"
-			rm -r "$_qmdir" || exit
+			rm -rf "$_qmdir" || exit
 		fi
 	done
 
@@ -66,6 +66,8 @@ install_qmail()
 	stage_exec ln -s /usr/local/vpopmail/qmail-users /var/qmail/users
 
 	mkdir -p "$STAGE_MNT/usr/local/etc/rc.d"
+
+	tell_status "setting qmail hostname to $TOASTER_HOSTNAME"
 	echo "$TOASTER_HOSTNAME" > "$ZFS_DATA_MNT/vpopmail/qmail-control/me"
 
 	stage_make_conf mail_qmail_ 'mail_qmail_SET=DNS_CNAME DOCS MAILDIRQUOTA_PATCH

--- a/provision-dns.sh
+++ b/provision-dns.sh
@@ -14,12 +14,24 @@ install_unbound()
 
 get_mt6_data()
 {
+	get_public_ip
+
+	local _spf_ips
+
+	if [ -z "$PUBLIC_IP6" ]; then
+		_spf_ips="ip4:$PUBLIC_IP4"
+	else
+		_spf_ips="ip4:$PUBLIC_IP4 ip6:$PUBLIC_IP6"
+	fi
+
 	echo "
 
 	   local-data: \"stage		A $(get_jail_ip stage)\"
 	   local-data: \"$(get_reverse_ip stage) PTR stage\"
 	   local-data: \"$TOASTER_HOSTNAME A $(get_jail_ip vpopmail)\"
-	   local-data: \"$TOASTER_HOSTNAME AAAA $(get_jail_ip6 vpopmail)\""
+	   local-data: \"$TOASTER_HOSTNAME AAAA $(get_jail_ip6 vpopmail)\"
+	   local-data: \"$TOASTER_HOSTNAME TXT 'v=spf1 a $_spf_ips -all'\"
+	   local-data: \"$TOASTER_MAIL_DOMAIN TXT 'v=spf1 a mx $_spf_ips -all'\""
 
 	for _j in $JAIL_ORDERED_LIST
 	do

--- a/provision-dns.sh
+++ b/provision-dns.sh
@@ -17,7 +17,9 @@ get_mt6_data()
 	echo "
 
 	   local-data: \"stage		A $(get_jail_ip stage)\"
-	   local-data: \"$(get_reverse_ip stage) PTR stage\""
+	   local-data: \"$(get_reverse_ip stage) PTR stage\"
+	   local-data: \"$TOASTER_HOSTNAME A $(get_jail_ip vpopmail)\"
+	   local-data: \"$TOASTER_HOSTNAME AAAA $(get_jail_ip6 vpopmail)\""
 
 	for _j in $JAIL_ORDERED_LIST
 	do

--- a/provision-haraka.sh
+++ b/provision-haraka.sh
@@ -610,6 +610,10 @@ configure_haraka()
 		echo '3' > "$HARAKA_CONF/tarpit.timeout"
 	fi
 
+	if [ ! -f "$HARAKA_CONF/me" ]; then
+		echo "$TOASTER_HOSTNAME" > "$HARAKA_CONF/me"
+	fi
+
 	if [ ! -f "$HARAKA_CONF/deny_includes_uuid" ]; then
 		echo '12' > "$HARAKA_CONF/deny_includes_uuid"
 	fi

